map_model/src/make/turns.rs

```diff
@@ -294,3 +297,67 @@ fn lc_penalty(t: &Turn, map: &Map) -> isize {
 
     ((from_idx as isize) - (to_idx as isize)).abs()
 }
+
+fn remove_merging_turns(map: &Map, input: Vec<Turn>, turn_type: TurnType) -> Vec<Turn> {
+    let mut turns = Vec::new();
+
+    // Group turns of the specified type by (from, to) road
+    let mut pairs: BTreeMap<(RoadID, RoadID), Vec<Turn>> = BTreeMap::new();
+    for t in input {
+        // Only do this for driving lanes
```

Comment:
As the videos in the PR show, there's still funniness with turns leading to/from bike/bus lanes. Should handle that next, but hopefully only after some kind of unit testing is in place.